test/test_torch.py

```diff
@@ -5343,6 +5344,117 @@ def test_potrs(self):
+            x_exp = torch.potrs(b.squeeze(0), L.squeeze(0), upper=upper)
+            x = torch.potrs(b, L, upper=upper)
+            self.assertEqual(x, x_exp.unsqueeze(0))
+
+            # test against potrs in a loop: four batches with both choices of upper
+            A = cast(random_symmetric_pd_matrix(5, 4))
+            L = get_cholesky(A, upper)
+            b = cast(torch.randn(4, 5, 10))
+
+            x_exp_list = list()
+            for i in range(4):
+                x_exp = torch.potrs(b[i], L[i], upper=upper)
+                x_exp_list.append(x_exp)
+            x_exp = torch.stack(x_exp_list)
+
+            x = torch.potrs(b, L, upper=upper)
+            self.assertEqual(x, x_exp)
+
+            # basic correctness test
+            A = cast(random_symmetric_pd_matrix(5, 3))
+            L = get_cholesky(A, upper)
+            b = cast(torch.randn(3, 5, 10))
+            x = torch.potrs(b, L, upper)
+            self.assertLessEqual(b.dist(torch.matmul(A, x)), 1e-12)
+
+            # Test non-contiguous inputs.
+            if not TEST_NUMPY:
+                return
+            import numpy
+            from numpy.linalg import solve
+            A = cast(random_symmetric_pd_matrix(2, 2)).permute(0, 2, 1)
+            b = cast(torch.randn(2, 2, 2)).permute(2, 1, 0)
+            L = get_cholesky(A, upper)
+            x = torch.potrs(b, L, upper=upper)
+            x_exp = torch.Tensor(solve(A.cpu().numpy(), b.cpu().numpy()))
+            self.assertEqual(x.data, cast(x_exp))
+
+    @skipIfNoLapack
+    def test_potrs_batched(self):
+        self._test_potrs_batched(self, lambda t: t)
+
+    @staticmethod
+    def _test_potrs_batched_dims(self, cast):
+        if not TEST_NUMPY:
+            return
+
+        from numpy.linalg import solve
+        from common_utils import random_symmetric_pd_matrix
+
+        for upper in [True, False]:
+            # TODO: This function should be replaced after batch potrf is ready
+            def get_cholesky(bmat, upper):
+                n = bmat.size(-1)
+                cholesky = torch.stack([m.cholesky(upper) for m in bmat.reshape(-1, n, n)])
+                return cholesky.reshape_as(bmat)
+
+            # test against numpy.linalg.solve
+            A = cast(random_symmetric_pd_matrix(4, 2, 1, 3))
+            b = cast(torch.randn(2, 1, 3, 4, 6))
+            L = get_cholesky(A, upper)
+            x = torch.potrs(b, L, upper=upper)
+            x_exp = torch.Tensor(solve(A.cpu().numpy(), b.cpu().numpy()))
+            self.assertEqual(x.data, cast(x_exp))
+
+            # broadcasting b
+            A = cast(random_symmetric_pd_matrix(4, 2, 1, 3))
+            b = cast(torch.randn(4, 6))
+            L = get_cholesky(A, upper)
+            x = torch.potrs(b, L, upper=upper)
+            x_exp = torch.Tensor(solve(A.cpu().numpy(), b.cpu().numpy()))
+            self.assertEqual(x.data, cast(x_exp))
+
+            # broadcasting A
+            A = cast(random_symmetric_pd_matrix(4))
```

Comment:
We can probably factor out a lot of this logic into a helper function,
test_batch_potrs_helper(A_cpu, b_cpu, tensor_cast, upper)

Reply:
Sure, I'll get that done. There might be some other tests that are deserving of such deduplication, so should I sort them out here, or create another PR?

Reply:
Either way works. It's probably nicer in a separate PR